src/dune_engine/scheduler.ml

```diff
@@ -847,23 +848,10 @@ end = struct
       t.handler t.config Tick;
       match Event.Queue.next t.events with
       | Job_completed (job, status) -> Fiber.Fill (job.ivar, status)
-      | Files_changed changed_files -> (
-        (* CR-someday amokhov: In addition to tracking files, we also need to
-           track directory listings. Otherwise, when a new file is added to a
-           source directory, [invalidate] will return [Skipped] because that
-           file was previously untracked. To fix this, we will need to introduce
-           new [Memo] cells for tracking directory listings, and [invalidate]
-           those cells when the listings change. *)
-        (* If none of [changed_files] is tracked by the build system, we will
-           ignore this [Files_changed] event to avoid unnecessary restarts. *)
-        let all_changed_files_skipped =
-          List.for_all changed_files ~f:(fun path ->
-              match Fs_memo.invalidate path with
-              | Skipped -> true
-              | Invalidated -> false)
-        in
-        match (all_changed_files_skipped, Memo.incremental_mode_enabled) with
-        | true, true -> iter t (* Ignore the event *)
+      | File_system_changed events -> (
+        let rebuild_required = Fs_memo.process_events events in
+        match (rebuild_required, Memo.incremental_mode_enabled) with
+        | No, true -> iter t (* Ignore the event *)
```

Comment:
I think `process_events` takes `incremental_mode_enabled` into account, but here we're doing this again.

Is there a reason to do it twice, or should we simplify one of them?

Reply:
No, I simply forgot to remove it from here. Well spotted!